### PR TITLE
Dfar.frontend

### PR DESCRIPTION
**Pull request by jenningsnickolas**
Regex now applied, ali's changing to progressive loading bar now added

### CODE REVIEW

File: xby2.apps.web/AngularApp/app/components/home/homeController.js [hunk 1]
```diff
@@ -6,24 +6,60 @@ define(["angular",
     function (angular, moment, lodash, testRepo, activityRepo, projectRepo) {
 
     var homeController = function ($scope) {
-        $scope.problemData = "Star Wars\n" +
-            "Star Wars Episode V\n" +
-            "Star Wars Episode VI\n" +
-            "Star Wars Episode I\n" +
-            "Star Wars Episode II\n" +
-            "Star Wars Episode III";
+        /* needs review, may be better
```
Comment (JeffSallans) [inline xby2.apps.web/AngularApp/app/components/home/homeController.js]: Remove commented out code

Comment (JeffSallans) [inline xby2.apps.web/AngularApp/app/components/home/homeController.js]: Add a todo comment and mention this is the object expected from the javascript repo.

File: xby2.apps.web/AngularApp/app/components/home/homeController.js [hunk 2]
```diff
@@ -6,24 +6,60 @@ define(["angular",
...
+        $scope.problemObject = {
+            "dataPreRegex": "Star Wars\n" +
```
Comment (JeffSallans) [inline xby2.apps.web/AngularApp/app/components/home/homeController.js]: no quotes for the first value

File: xby2.apps.web/AngularApp/app/components/home/homeController.js [hunk 3]
```diff
@@ -6,24 +6,60 @@ define(["angular",
...
+            return {
+                "width": $scope.progress + "%"
+            };
+        };
+
+        $scope.updateProgress = function() {
+            if ($scope.progress < 100) {
+                //TODO: This number will be changed when we know the number of REGEX puzzles
+                $scope.progress += 10;
+                $scope.nextButtonDisabled = 1;
```
Comment (JeffSallans) [inline xby2.apps.web/AngularApp/app/components/home/homeController.js:21]:
``` javascript
$scope.nextButtonDisabled = true;  
//$scope.nextButtonDisabled => true
```

File: xby2.apps.web/AngularApp/app/components/home/homeController.js [hunk 4]
```diff
@@ -6,24 +6,60 @@ define(["angular",
...
+            var regexInternal = new RegExp($scope.regex, 'g');
+            $scope.filteredResults = $scope.problemData.match(regexInternal).toString().replace(/,/g, "\n");
+            $scope.filteredResults == $scope.answers ? $scope.nextButtonDisabled = 0 : $scope.nextButtonDisabled = 1;
         };
+
+        $scope.problemMetadata = {
+            "number": "1",
+            "difficulty": "1",
+            "title":"Star Wars only the bad ones."
+        }
```
Comment (JeffSallans) [inline xby2.apps.web/AngularApp/app/components/home/homeController.js]: add ;

File: xby2.apps.web/AngularApp/app/components/home/homeController.js [hunk 5]
```diff
@@ -6,24 +6,60 @@ define(["angular",
...
 
+        $scope.answers = "Star Wars Episode I\nStar Wars Episode II\nStar Wars Episode III";
+
+        $scope.filteredResults = $scope.problemData;
+
+        $scope.regex="";
+        
         $scope.regexChange = function () {
-            alert('hello world');
+            var regexInternal = new RegExp($scope.regex, 'g');
```
Comment (JeffSallans) [inline xby2.apps.web/AngularApp/app/components/home/homeController.js]: We should add a try catch to handle this

File: xby2.apps.web/AngularApp/app/components/home/homeController.js (hunk 3 above)
Comment (dfar-io) [inline xby2.apps.web/AngularApp/app/components/home/homeController.js:21]: All set

**Pull request merged**